optaplanner-examples/src/main/java/org/optaplanner/examples/investment/solver/move/factory/InvestmentQuantityTransferMoveIteratorFactory.java

```diff
@@ -88,7 +90,7 @@ public boolean hasNext() {
         }
 
         @Override
-        public Move next() {
+        public Move<InvestmentSolution> next() {
```

Comment:
Yes we can. There is just a minor side effect with "Exporting non-public type through public API". So I'll just make the iterator class public (because it will be effectively public anyway).